app/src/actions/tasks.ts

```diff
@@ -31,15 +33,24 @@ export function loadTasks() {
 export function addTask(task: Task, onComplete: Function) {
   return (dispatch) => {
     delete task._id;
-    return Tasks.add(task)
-      .subscribe((res) => {
-        if (res.length === 1) {
-          dispatch({
-            type: TASK_ADDED,
-            payload: res[0]
-          });
-        }
-      }, null, onComplete); }
+
+    const taskAdd$ = Tasks.add(task).publish();
+    taskAdd$.connect();
+
+    taskAdd$.subscribe(res=> {
+      if (res.length === 1) {
+        dispatch({
+          type: TASK_ADDED,
+          payload: res[0]
+        });
+      }
+    })
+
+
+    return taskAdd$;
+
+
+  }
```

Comment:
So I'm thinking about this solution, and it seems like it should work fine - but there may be a more elegant way about it - if I am not making any mistakes about the relationship between actions and reducers. It would also mean 

First idea - probably the least amount of change and entirely superficial:

```javascript

taskAdd$
.filter(res => res.length === 1)
.pluck('0') //Probably gratuitous
.subscribe(res => dispatch({type: TASK_ADDED, payload: res}); 
return taskAdd$;
```

You can see, not super essential, but maybe looks a bit nicer and is a little more functionally declarative.

The more interesting change (in my opinion) would be having the reducer subscribe to the task observables - but the more I think about it, the more I realize just how complex that would be. Still interesting, but I think something that could potentially be explored with the ng2ReduRx re-write.